### PR TITLE
Configure trackings per site

### PR DESCRIPTION
**Pull request by fvovan**
Allow to configure `tracking` service by site.

- Add $siteName to params of CM_Service_Tracking_ClientInterface methods
which doesn't accept `$environment` for now

### CODE REVIEW

**Comment by fvovan** [comment]
I have one more question.
What we are going to do in case of tracking things which doesn't belong to any site at the moment of their  tracking? For example in `CM_Action_Abstract::_track()` (so far) , should we extract 'actions trackers' to another service, which will be common for all sites of the app?
wdyt @fauvel @njam ?

**Comment by fvovan** [comment]
Second case is `CM_Model_Splittest::_getVariationFixture()`. Even though it's possible to pass SiteId when we process `CM_Model_Splittest_RequestClient`, but `CM_Model_Splittest_User` does not (and most likely should not) contain any relations to `Site`.
So I propose to detach action and split test trackings to separate interfaces and services.

File: library/CM/Class/Abstract.php [hunk 1]
```diff
@@ -22,6 +22,13 @@ public function getClassHierarchy() {
     }
 
     /**
+     * @return string[] List of class names
+     */
+    public static function getClassHierarchyStatic() {
+        return self::_getClassHierarchy();
```
Comment (fauvel) [inline library/CM/Class/Abstract.php]: This doesn't make much sense imo – you could directly make `getClassHierarchy` static if needed.

Comment (fvovan) [inline library/CM/Class/Abstract.php]: yes, but sometimes it's used in tests where it's mocked as instance method so I kept it for more compatibility.

**Comment by njam** [comment]
@fvovan @fauvel closeable?

**Pull request closed without merging**